Source/src_android/PinCheckerLibrary/src/main/java/com/wultra/android/passphrasemeter/PasswordTester.java

```diff
@@ -150,13 +158,30 @@ public PinTestResult testPin(@NonNull String pin) throws WrongPinException {
             }
         }
 
-        return new PinTestResult(pin, set);
+        return new PinTestResult(pinBytes.length, set);
     }
 
     // Private methods & constants
 
     /**
-     * Defines constants returned from {@link #testPasswordJNI(String)} method.
+     * Converts user's byte input into ByteBuffer
+     *
+     * @param byteArray Password or PIN as byte array
+     * @param arrayLength length of an array
+     * @return Shared byte buffer between Java/JNI
+     */
+    private ByteBuffer getSharedBuffer(byte[] byteArray, int arrayLength) {
+        ByteBuffer bb = ByteBuffer.allocateDirect(arrayLength);
```

Comment:
I think that usage of ByteBuffer here is too heavy. We don't transfer big memory regions, so there's no need to share exactly the same memory region between JVM and the native code.

I would keep `byte[]` as a type passed to JNI. What we need is to take a special care when JVM does the copy of the buffer into the native code space. The `GetByteArrayElements()` function has `is_copy` parameter that tells us that array was copied, so the temporary buffer has to be erased.